Automattic/jetpack
# [RNMobile] Add Privacy and Rating panel to VideoPress Block Settings

> **Warning**
> This PR's main focus is to add UI for the Privacy and Rating settings. Functionally, changes to these settings won't synchronise as expected across the web and the app for now. This will be addressed in later PRs.

Part of an ongoing project to add the VideoPress block to the native apps.

## Proposed changes:

* A native version of the `PrivacyAndRatingPanel` component has been introduced and added to the VideoPress block's settings. 
* The code specific to the panel's settings is shared across both the web and native. As such, it's been extracted into a new component in https://github.com/Automattic/jetpack/pull/29477/commits/43b954ff643a88b63980780bc31b337fc9f1823f, so that it can be used across both platforms.
* Tweaks to the settings in this panel won't be correctly synchronised with the web with this PR. The code needed for such synchronisation will be added separately in later PR 

### Other information:


## Jetpack product discussion

p9ugOq-3mV-p2

## Does this pull request change what data or activity we track or use?

No, it does not.

## Testing instructions:

* Navigate to an existing VideoPress block or go through the steps to add a new one.
* Tap the settings toggle to verify and then tap `Privacy and Rating` to open the new settings panel.
* Experiment with toggling the new settings on/off and ensuring they visually work as expected.

### Screenshots

| Main sheet  | Subsheet |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/226974732-5b615602-32ec-45c4-8ee5-8f28513fcb4d.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/226424079-5d0fba48-86bf-447f-8b7d-a4d5a53c6705.png" width="100%"> | 

### Screen Recording 

https://user-images.githubusercontent.com/2998162/226975042-6abfd531-9401-4f22-a6cc-e597a5730e11.mov
